src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -246,6 +237,79 @@ public static function register($mode = 0)
         }
     }
 
+    private static function isLaggingVendor(array $trace): bool
+    {
+        foreach ($trace as $line) {
+            if (isset($line['file'])) {
+                $file = $line['file'];
+                if ('-' === $file) {
+                    continue;
+                }
+                if (!self::inVendors($file)) {
+                    return false;
+                }
+                if (isset($erroringFile, $erroringPackage)) {
```

Comment:
would be better to initialize variables to `null` before, and compare here instead. Having code dealing with undefined variables without failing makes it much easier to introduce mistakes (making a typo in a variable name does not trigger an error anymore)